turbine/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -268,7 +268,18 @@ impl StandardBroadcastRun {
 
         // 2) Convert entries to shreds and coding shreds
         let is_last_in_slot = last_tick_height == bank.max_tick_height();
-        let reference_tick = bank.tick_height() % bank.ticks_per_slot();
+        // PoH tick_height is cumulative across all slots since genesis. On the
+        // other hand, reference_tick is an index for how many of THIS slot's
+        // ticks have occurred already. The conversion is performed below,
+        // including saturating_sub(1) to map from 1-based to 0-based indexing.
+        //
+        // As of the writing of this comment, reference_tick is allocated 6
+        // bits in the shred header. So, a ticks_per_slot value greater than 2^6
+        // (64) cannot be properly encoded and may cause undesired side effects
+        let reference_tick = last_tick_height
+            .saturating_add(bank.ticks_per_slot())
+            .saturating_sub(bank.max_tick_height())
+            .saturating_sub(1);
```

Comment:
If I'm parsing this correctly, it looks like every ref tick will be -1 compared to pre-change values with the exception of:
* tick 0 saturates at 0
* last tick will be 63 (instead of 0)

That seems reasonable to me given the margin we apply to this during repair

Reply:
> Edit: from your attached graph, looks like tick starts at 1, but not sure if this is just the replay/PoH/Banking Stage/Broadcast handoff delay

Yeah, the scale makes it hard to see, but the minimum value was actually `reference_tick = 3`. I only looked at that one block so I'm not sure if that was a "this block", "this leader", "this version", etc issue. But, still good to consider all possibilities so that changes / optimizations somewhere else are less likely to cause problems here

Reply:
Marking this conversation as resolved as we're not toggling the code it references back. There is some good information in https://github.com/anza-xyz/agave/pull/4644#discussion_r1931363353.

Also, I added a sanity check (which I later removed) in https://github.com/anza-xyz/agave/pull/4644/commits/a7ce217f1bdd209f3909cdb188ebcfe3a7cab25d to show that the logic in shred will handle mapping `64 ==> 63` as long as we don't map `64 ==> 0` first 😅